ocaml/xapi/xapi_vm.ml

```diff
@@ -279,10 +279,28 @@ let assert_memory_constraints ~__context ~vm platformdata =
     let c = C.get ~__context ~vm_ref:vm in
     C.assert_valid_and_pinned_at_static_max ~constraints:c ~reason:"SRIOV vGPU"
 
-(* Note: it is important that we use the pool-internal API call, VM.atomic_set_resident_on, to set resident_on and clear
-   scheduled_to_be_resident_on atomically. This prevents concurrent API calls on the master from accounting for the
-   same VM twice during memory calculations to determine whether a given VM can start on a particular host..
-*)
+let enforce_memory_constraints ~__context ~vm _platformdata =
+  let module C = Xapi_vm_memory_constraints.Vm_memory_constraints in
+  let constraints = C.get ~__context ~vm_ref:vm in
+  try
+    C.assert_valid_and_pinned_at_static_max ~constraints
+      ~reason:"DMC unavailable"
+  with Api_errors.Server_error _ ->
```

Comment:
Could we use `are_valid_and_pinned_at_static_max` directly so we don't need to raise/catch exceptions?